receiver/statsdreceiver/factory.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statsdreceiver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configerror"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr             = "statsd"
+	defaultBindEndpoint = "localhost:8125"
+	defaultTransport    = "udp"
+)
+
+// Factory is the factory for StatsD receiver.
+type Factory struct {
+}
+
+var _ component.ReceiverFactoryOld = (*Factory)(nil)
```

Comment:
component.ReceiverFactoryOld was removed, please use the receiverhelper (see carbon or other receivers as examples).

Reply:
@sonofachamp sorry for the churn ><

Reply:
All good, this is fixed.